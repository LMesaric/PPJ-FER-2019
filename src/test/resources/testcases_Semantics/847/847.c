int main(void) {
  char d[10]={'a','b','c'};
  char dd[10]={'d','e','f'};
  d=dd;
  return 0;
}
