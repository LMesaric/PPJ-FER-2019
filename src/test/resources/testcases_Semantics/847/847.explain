semantic: d is not l-value
note: 'd = dd' on line 4
