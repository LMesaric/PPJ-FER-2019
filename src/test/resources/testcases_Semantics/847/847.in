<prijevodna_jedinica>
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 1 int
   IDN 1 main
   L_ZAGRADA 1 (
   KR_VOID 1 void
   D_ZAGRADA 1 )
   <slozena_naredba>
    L_VIT_ZAGRADA 1 {
    <lista_deklaracija>
     <lista_deklaracija>
      <deklaracija>
       <ime_tipa>
        <specifikator_tipa>
         KR_CHAR 2 char
       <lista_init_deklaratora>
        <init_deklarator>
         <izravni_deklarator>
          IDN 2 d
          L_UGL_ZAGRADA 2 [
          BROJ 2 10
          D_UGL_ZAGRADA 2 ]
         OP_PRIDRUZI 2 =
         <inicijalizator>
          L_VIT_ZAGRADA 2 {
          <lista_izraza_pridruzivanja>
           <lista_izraza_pridruzivanja>
            <lista_izraza_pridruzivanja>
             <izraz_pridruzivanja>
              <log_ili_izraz>
               <log_i_izraz>
                <bin_ili_izraz>
                 <bin_xili_izraz>
                  <bin_i_izraz>
                   <jednakosni_izraz>
                    <odnosni_izraz>
                     <aditivni_izraz>
                      <multiplikativni_izraz>
                       <cast_izraz>
                        <unarni_izraz>
                         <postfiks_izraz>
                          <primarni_izraz>
                           ZNAK 2 'a'
            ZAREZ 2 ,
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          ZNAK 2 'b'
           ZAREZ 2 ,
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         ZNAK 2 'c'
          D_VIT_ZAGRADA 2 }
       TOCKAZAREZ 2 ;
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_CHAR 3 char
      <lista_init_deklaratora>
       <init_deklarator>
        <izravni_deklarator>
         IDN 3 dd
         L_UGL_ZAGRADA 3 [
         BROJ 3 10
         D_UGL_ZAGRADA 3 ]
        OP_PRIDRUZI 3 =
        <inicijalizator>
         L_VIT_ZAGRADA 3 {
         <lista_izraza_pridruzivanja>
          <lista_izraza_pridruzivanja>
           <lista_izraza_pridruzivanja>
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          ZNAK 3 'd'
           ZAREZ 3 ,
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         ZNAK 3 'e'
          ZAREZ 3 ,
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        ZNAK 3 'f'
         D_VIT_ZAGRADA 3 }
      TOCKAZAREZ 3 ;
    <lista_naredbi>
     <lista_naredbi>
      <naredba>
       <izraz_naredba>
        <izraz>
         <izraz_pridruzivanja>
          <postfiks_izraz>
           <primarni_izraz>
            IDN 4 d
          OP_PRIDRUZI 4 =
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        IDN 4 dd
        TOCKAZAREZ 4 ;
     <naredba>
      <naredba_skoka>
       KR_RETURN 5 return
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      BROJ 5 0
       TOCKAZAREZ 5 ;
    D_VIT_ZAGRADA 6 }
