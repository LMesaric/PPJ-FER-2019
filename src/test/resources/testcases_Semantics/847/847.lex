KR_INT 1 int
IDN 1 main
L_ZAGRADA 1 (
KR_VOID 1 void
D_ZAGRADA 1 )
L_VIT_ZAGRADA 1 {
KR_CHAR 2 char
IDN 2 d
L_UGL_ZAGRADA 2 [
BROJ 2 10
D_UGL_ZAGRADA 2 ]
OP_PRIDRUZI 2 =
L_VIT_ZAGRADA 2 {
ZNAK 2 'a'
ZAREZ 2 ,
ZNAK 2 'b'
ZAREZ 2 ,
ZNAK 2 'c'
D_VIT_ZAGRADA 2 }
TOCKAZAREZ 2 ;
KR_CHAR 3 char
IDN 3 dd
L_UGL_ZAGRADA 3 [
BROJ 3 10
D_UGL_ZAGRADA 3 ]
OP_PRIDRUZI 3 =
L_VIT_ZAGRADA 3 {
ZNAK 3 'd'
ZAREZ 3 ,
ZNAK 3 'e'
ZAREZ 3 ,
ZNAK 3 'f'
D_VIT_ZAGRADA 3 }
TOCKAZAREZ 3 ;
IDN 4 d
OP_PRIDRUZI 4 =
IDN 4 dd
TOCKAZAREZ 4 ;
KR_RETURN 5 return
BROJ 5 0
TOCKAZAREZ 5 ;
D_VIT_ZAGRADA 6 }
