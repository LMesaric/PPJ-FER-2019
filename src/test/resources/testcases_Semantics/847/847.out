<izraz_pridruzivanja> ::= <postfiks_izraz> OP_PRIDRUZI(4,=) <izraz_pridruzivanja>
